backend/coins/btc/account.go

```diff
@@ -476,26 +475,72 @@ func (account *Account) Notifier() accounts.Notifier {
 	return account.notifier
 }
 
-func (account *Account) updateFeeTargets() {
-	defer account.feeTargetsLock.Lock()()
+// feeTargets fetches the available fees. For mainnet BTC it uses mempool.space estimation.
+//
+// For the other coins or in case mempool.space is not available it fallbacks on Bitcoin Core.
+// The minimum relay fee is used as a last resource fallback in case also Bitcoin Core is
+// unavailable.
+func (account *Account) feeTargets() []*FeeTarget {
+	// for mainnet BTC we fetch mempool.space fees, as they should be more reliable.
+	mempoolFees := &accounts.MempoolSpaceFees{}
+	if account.coin.Code() == coin.CodeBTC {
+		_, err := util.APIGet(account.httpClient, mempoolSpaceMirror, "", 1000, mempoolFees)
+		if err != nil {
+			mempoolFees = nil
+			account.log.WithError(err).Error("Fecthing fees from " + mempoolSpaceMirror + " failed")
+		}
+	}
+
+	// feeTargets must be sorted by ascending priority.
+	var feeTargets []*FeeTarget
+	if mempoolFees != nil {
+		feeTargets = []*FeeTarget{
+			{blocks: 12, code: accounts.FeeTargetCodeMempoolEconomy},
+			{blocks: 3, code: accounts.FeeTargetCodeMempoolHour},
+			{blocks: 2, code: accounts.FeeTargetCodeMempoolHalfHour},
+			{blocks: 1, code: accounts.FeeTargetCodeMempoolFastest},
+		}
+	} else {
+		feeTargets = []*FeeTarget{
+			{blocks: 24, code: accounts.FeeTargetCodeEconomy},
+			{blocks: 12, code: accounts.FeeTargetCodeLow},
+			{blocks: 6, code: accounts.FeeTargetCodeNormal},
+			{blocks: 2, code: accounts.FeeTargetCodeHigh},
+		}
+	}
+
 	var minRelayFeeRate *btcutil.Amount
 	minRelayFeeRateVal, err := account.getMinRelayFeeRate()
 	if err == nil {
 		minRelayFeeRate = &minRelayFeeRateVal
 	}
-	for _, feeTarget := range account.feeTargets {
-		feeRatePerKb, err := account.coin.Blockchain().EstimateFee(feeTarget.blocks)
-		if err != nil {
-			if account.coin.Code() != coin.CodeTLTC {
+
+	for _, feeTarget := range feeTargets {
+		var feeRatePerKb btcutil.Amount
+
+		if mempoolFees != nil {
```

Comment:
this condition is true always even for TBTC or LTC as you initialize it to ` mempoolFees := &accounts.MempoolSpaceFees{}`.

Reply:
Shoot! I don't know how I missed this during the testing. Great catch :pray: